go.mod

```diff
@@ -1,13 +1,15 @@
 module github.com/bufbuild/buf
 
-go 1.20
+go 1.21
```

Comment:
Why is this happening? @emcfarlane

Reply:
I think from the command it removes the files and recreates. 

```
$ make upgrade
rm -f go.mod go.sum
go mod init github.com/bufbuild/buf
```

Reply:
Its due to otel: `go.opentelemetry.io/otel@v1.25.0 declares Go version 1.21`

https://github.com/open-telemetry/opentelemetry-go/blob/v1.25.0/go.mod

Reply:
Ugh. We need to keep v1.20 support for now. Can you add an explicit pin to https://github.com/bufbuild/buf/blob/main/make/buf/all.mk#L2

```
GO_GET_PKGS := $(GO_GET_PKGS) go.opentelemetry.io/otel@v1.24.0
```

Reply:
I think you'll have to add `sdk` and `trace` too:

```
GO_GET_PKGS := $(GO_GET_PKGS) \
  go.opentelemetry.io/otel@v1.24.0 \
  go.opentelemetry.io/otel/sdk@v1.24.0 \
  go.opentelemetry.io/otel/trace@v1.24.0
```

Reply:
And put a comment `TODO: Remove when we no longer need to support Golang <1.21`

Reply:
Should be fixed now.